lib/src/room.dart

```diff
@@ -274,18 +274,22 @@ class KurentoRoom extends ChangeNotifier implements Room {
 
   @override
   Future<void> replaceStream(MediaStream mediaStream) async {
-    // Replace the stored local stream.
-    _localStream = mediaStream;
-
-    // Replace the tracks.
-    await _connectionByTrackId[_localTrackId]!.replaceAudioTrack(
-        _isAudioMuted || mediaStream.getAudioTracks().isEmpty ? null : mediaStream.getAudioTracks()[0]);
-    if (!_isPresenting) {
-      await _connectionByTrackId[_localTrackId]!
-          .replaceVideoTrack(_isVideoMuted || !_hasVideoTrack ? null : mediaStream.getVideoTracks()[0]);
-    }
+    if (_serviceRequestState == ServiceRequestState.queued) {
```

Comment:
This doesn't seem like the right fix, since what we really need to check for is whether the participant has joined the room. If the participant has not joined the room yet, they shouldn't be allowed to replace the stream.

How about:
```dart
if (_localStream != null) {
```

Reply:
I'd prefer to use `_localTrackId`